lang/cs/Org.Apache.REEF.Client/API/DriverStatus.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Client.API
+{
+    /// <summary>
+    /// Represents the status of a Driver.
+    /// </summary>
+    internal enum DriverStatus
+    {
+        /// <summary>
+        /// Driver is initializing.
+        /// </summary>
+        INIT,
+
+        /// <summary>
+        /// Driver is running.
+        /// </summary>
+        RUNNING,
+
+        /// <summary>
+        /// Driver has cleanly exited.
+        /// </summary>
+        DONE,
+
+        /// <summary>
+        /// Driver is suspended.
+        /// </summary>
+        SUSPEND,
+
+        /// <summary>
+        /// Driver was killed.
+        /// </summary>
+        KILLED,
+
+        /// <summary>
+        /// Driver Failed.
+        /// </summary>
+        FAILED,
+
+        /// <summary>
+        /// Driver was RUNNING, but is no longer reachable for unknown reasons.
+        /// </summary>
+        UNKNOWN_EXITED
```

Comment:
probably, `NOT_AVAILABLE`?

Reply:
We already have ApplicationState. Driver in REEF essentially is the application. Would it cause confusion if we introduce another states.

Reply:
Where are those status set for driver? In Java? Then should the status match the DriverStatus defined in Java?

Reply:
Hmm, that doesn't convey the status I meant. Maybe `NO_LONGER_AVAILABLE` to indicate that the status of this Driver used to be available?

Reply:
@jwang98052 said:

>We already have ApplicationState. Driver in REEF essentially is the application. Would it cause confusion if we introduce another states.

Hmm, ApplicationState is a YARN concept and distinct from the REEF concept, so it should be fine. That being said, this really should be thought through if / when we build a REEF.NET client.

Reply:
@jwang98052 said:
>Where are those status set for driver? In Java? Then should the status match the DriverStatus defined in Java?

Yes, they are set in Java, in the new class `DriverStatusHTTPHandler` introduced in this PR. That class implements `JobStatusHandler`, the interface used within the Driver to communicate the Driver (job) status to the client. It in turn relies on `ReefServiceProtos.JobStatusProto` as the representation of the Job Status. That class is generated from the protobuf spec in `lang/common/proto/reef_service_protos.proto`, which defines `State` as:

```proto
enum State {
    INIT = 0;
    RUNNING = 1;
    DONE = 2;
    SUSPEND = 3;
    FAILED = 4;
    KILLED = 5;
}
```

This is the source why I used `SUSPEND`, not `SUSPENDED`. If we don't change the proto, but want to change the code, we need logic that maps one to the other. That would introduce more potential bugs than it seems to be worth it.